homeassistant/components/cover/__init__.py

```diff
@@ -251,7 +252,7 @@ def current_cover_tilt_position(self) -> int | None:
         """
         return self._attr_current_cover_tilt_position
 
-    @property
+    @cached_property
     def device_class(self) -> CoverDeviceClass | None:
```

Comment:
This will not work, as the signature is different now in all implementations.

Reply:
Right. The typing on cached_property is too strict.  It needs to be changed to work the same as property

Reply:
Yup... ran into the same :) I tried to make this exact same change before.

../Frenck

Reply:
Adjusted... still not 100% sure its right though